agent/agent_worker.go

```diff
@@ -61,12 +68,22 @@ func (a AgentWorker) Create() AgentWorker {
 	}
 
 	a.APIClient = APIClient{Endpoint: endpoint, Token: a.Agent.AccessToken}.Create()
-
 	return a
 }
 
 // Starts the agent worker
 func (a *AgentWorker) Start() error {
+	a.metrics = a.MetricsCollector.Scope(metrics.Tags{
+		"hostname":   a.Agent.Hostname,
```

Comment:
On datadog you (can) pay per metric, and tag cardinality can easily explode into many distinct metrics you're not careful. The datadog agent adds some [default tags](https://docs.datadoghq.com/tagging/) - including `host` - which are free. If this feature is specially aimed at DD users then this can safely be dropped, to save some users $$. Or adding these tags could be made configurable.

I'd also like to make the `agent_name` tag optional if possible, as for us it's the same as the kubernetes pod, which is also a free/reserved tag.

Reply:
We’re going to have a DD Buildkite integration, which means that these tags will all be free. Given that, is it still a concern?

Reply:
I've removed `hostname` @DazWorrall, thanks for the heads up.

Reply:
Thanks @lox 

> Given that, is it still a concern?

Nope, that helps with a bunch of things, thanks!